wire-ios-sync-engine/Source/Synchronization/ZMOperationLoop+PushChannel.swift

```diff
@@ -17,11 +17,33 @@
 //
 
 import Foundation
+import WireAPI
 
 extension ZMOperationLoop: ZMPushChannelConsumer {
 
-    public func pushChannelDidReceive(_ data: ZMTransportData) {
-        if let events = ZMUpdateEvent.eventsArray(fromPushChannelData: data), !events.isEmpty {
+    public func pushChannelDidReceive(_ data: Data) {
+        // TODO: WPB-9612 remove event decoding monitoring.
+        // Since update event models aren't documented, we aren't yet completely sure
+        // that the new type-safe UpdateEvent model can successfully decode all event
+        // types. To test correctness, we try to decode every update event received
+        // through the push channel and simply log when it fails so we know how to
+        // fix it. Once we're sure it works, we should remove this.
+        do {
+            let decoder = JSONDecoder.defaultDecoder
+            _ = try decoder.decode(UpdateEventEnvelope.self, from: data)
+        } catch {
+            WireLogger.updateEvent.error("failed to deserialize UpdateEventEnvelope: \(error)")
+        }
+
+        guard let transportData = try? JSONSerialization.jsonObject(
+            with: data,
+            options: []
+        ) as? ZMTransportData else {
+            WireLogger.updateEvent.error("failed to deserialize push channel data")
+            return
+        }
```

Comment:
why do you decode twice ?

Reply:
The first decoding is to the new event models, the second decoding is the existing event models in the form of `[AnyHashable: Any]`. These models aren't equivalent and our event processing only knows about the dictionaries.